apis/compute/v1beta1/gkecluster_types.go

```diff
@@ -0,0 +1,1260 @@
+
+	// Description: An optional description of this cluster.
+	// +optional
+	// +immutable
+	Description *string `json:"description,omitempty"`
+
+	// EnableKubernetesAlpha: Kubernetes alpha features are enabled on this
+	// cluster. This includes alpha API groups (e.g. v1alpha1) and features that
+	// may not be production ready in the kubernetes version of the master and
+	// nodes. The cluster has no SLA for uptime and master/node upgrades are
+	// disabled. Alpha enabled clusters are automatically deleted thirty days
+	// after creation.
+	// +optional
+	// +immutable
+	EnableKubernetesAlpha *bool `json:"enableKubernetesAlpha,omitempty"`
+
+	// EnableTpu: Enable the ability to use Cloud TPUs in this cluster.
+	// +optional
+	// +immutable
+	EnableTpu *bool `json:"enableTpu,omitempty"`
+
+	// InitialClusterVersion: The initial Kubernetes version for this
+	// cluster.  Valid versions are those
+	// found in validMasterVersions returned by getServerConfig.  The
+	// version can
+	// be upgraded over time; such upgrades are reflected
+	// in
+	// currentMasterVersion and currentNodeVersion.
+	//
+	// Users may specify either explicit versions offered by
+	// Kubernetes Engine or version aliases, which have the following
+	// behavior:
+	//
+	// - "latest": picks the highest valid Kubernetes version
+	// - "1.X": picks the highest valid patch+gke.N patch in the 1.X
+	// version
+	// - "1.X.Y": picks the highest valid gke.N patch in the 1.X.Y version
+	// - "1.X.Y-gke.N": picks an explicit Kubernetes version
+	// - "","-": picks the default Kubernetes version
+	// +optional
+	// +immutable
+	InitialClusterVersion *string `json:"initialClusterVersion,omitempty"`
+
+	// IpAllocationPolicy: Configuration for cluster IP allocation.
+	// +optional
+	// +immutable
+	IpAllocationPolicy *IPAllocationPolicy `json:"ipAllocationPolicy,omitempty"`
+
+	// LabelFingerprint: The fingerprint of the set of labels for this
+	// cluster.
+	// +optional
+	// +immutable
+	LabelFingerprint *string `json:"labelFingerprint,omitempty"`
+
+	// LegacyAbac: Configuration for the legacy ABAC authorization mode.
+	// NOTE(hasheddan): this can only be updated via setLegacyAbac
+	// https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters/setLegacyAbac
+	// +optional
+	LegacyAbac *LegacyAbac `json:"legacyAbac,omitempty"`
+
+	// Location: [Output only] The name of the Google Compute
+	// Engine
+	// [zone](/compute/docs/regions-zones/regions-zones#available)
+	// or
+	// [region](/compute/docs/regions-zones/regions-zones#available) in
+	// which
+	// the cluster resides.
+	// NOTE(hasheddan): this is labelled as Output Only by GCP but is required
+	// to create a cluster. It is not included in the actual cluster object
+	// itself, but is instead passed to the create call. If a region is given
+	// the cluster will be Regional, if a zone is given the cluster will be
+	// Zonal.
+	// +immutable
+	Location string `json:"location"`
```

Comment:
`Location` has been added to struct despite it being listed as `Output Only` because it is needed in the API call parameters, not the `*container.Cluster` object.